code/modules/power/generator.dm

```diff
@@ -152,6 +188,15 @@
 		else
 			disconnect_from_network()
 		reconnect()
+	if(istype(W, /obj/item/weapon/reagent_containers))
+		var/obj/item/weapon/reagent_containers/R = W
+		if(!R.reagents.has_reagent(/datum/reagent/lube/oil))
+			audible_message("<span class='warning'>[src] blips in disappointment</span>")
+			playsound(get_turf(src), 'sound/machines/synth_no.ogg', 50, 0)
+			return
+		R.standard_pour_into(user, src)
```

Comment:
You're only checking if the container _has_ oil in it for validation. This means you can put a drop of oil in a bucket of human blood and dump the concoction in the engine just fine. Which is pretty metal. 

Coupled with the other check I commented on, you can lubricate the engine with literally anything from water, through adminordazine, to molten iron.

Reply:
I can't see a decent way around this, although with the changes on the reagent check in process() i can just make the TEG dump any reagent that isn't oil.

Reply:
The way to do it is to `reagents.remove_reagent(/datum/reagent/lube/oil, amount)`, however, it'd be pretty odd if somehow only the oil was coming out of something you were pouring.